Mathlib/GroupTheory/Subsemigroup/Center.lean

```diff
@@ -154,22 +278,52 @@ variable (M)
 
 @[to_additive (attr := simp) addCenter_eq_univ]
 theorem center_eq_univ [CommSemigroup M] : center M = Set.univ :=
-  (Subset.antisymm (subset_univ _)) fun x _ y => mul_comm y x
+  (Subset.antisymm (subset_univ _)) (by
+    intros _ _
+    rw [_root_.Semigroup.mem_center_iff]
+    intro _
+    rw [mul_comm]
+    )
 #align set.center_eq_univ Set.center_eq_univ
 #align set.add_center_eq_univ Set.addCenter_eq_univ
 
 end Set
 
+namespace Mul
+
+section
+
+variable (M) [Mul M]
+
+/-- The center of a magma `M` is the set of elements that commute with everything in `M` -/
+@[to_additive
+      "The center of a magma `M` is the set of elements that commute with everything in `M`"]
+def center : Subsemigroup M where
+  carrier := Set.center M
+  mul_mem' := Set.mul_mem_center
+
+variable {M}
+
+/-- The center of a magma is commutative. -/
+@[to_additive "The center of an additive magma is commutative."]
+instance : CommSemigroup (Mul.center M) where
+  mul_assoc _ b _ := Subtype.ext <| b.2.mid_assoc _ _
+  mul_comm a _ := Subtype.ext <| a.2.comm _
+
+end
+
+end Mul
+
 namespace Subsemigroup
 
 section
 
-variable (M) [Semigroup M]
+variable (M)
 
 /-- The center of a semigroup `M` is the set of elements that commute with everything in `M` -/
 @[to_additive
       "The center of a semigroup `M` is the set of elements that commute with everything in `M`"]
-def center : Subsemigroup M where
+def center [Mul M] : Subsemigroup M where
```

Comment:
Isn't this now a duplicate of `Mul.center`? I think the `Subsemigroup.center` name is better, but indeed the docstring should change.

Reply:
Yes, I've removed `Mul.center`.